src/Bonsai.Scripting.Python/EnvironmentHelper.cs

```diff
@@ -2,31 +2,101 @@
 using System.IO;
 using System.Linq;
 using Python.Runtime;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text.RegularExpressions;
 
 namespace Bonsai.Scripting.Python
 {
     static class EnvironmentHelper
     {
-        public static string GetPythonDLL(string path)
+
+        private static string GetFirstFile(string path, string pattern, HashSet<string> visitedDirectories = null)
```

Comment:
This turns out to be a gigantic rabbit hole, but this solution unfortunately cannot guarantee that the correct python shared libraries will be found. The problem lies with [Multiarch](https://wiki.debian.org/Multiarch/LibraryPathOverview) handling of shared library paths.

Multiarch is the linux standard to allow side-by-side installation of 32-bit and 64-bit shared libraries in 64-bit systems which can run both types of applications. To do this, shared libraries with multiarch-support may be installed simultaneously to `x86_64-linux` or `i386-linux` folders inside `usr/lib` for example.

Because of folder sort order, systems with both python 64-bit and 32-bit installed may pick up the `i386-linux` folder accidentally simply because it shows up first in the hierarchical search.

I found this issue while looking for alternative ways to automatically discover `libpython3*.so` using only environment or system variables. Unfortunately I have no viable solution yet, but for a number of reasons related to both performance and security we should avoid doing an arbitrary recursive search on system and user folders.

Reply:
Another clue might be a symbolic link to the actual shared library stored in what seems a possibly predictable folder path: `/usr/lib/python3.10/config-3.10-x86_64-linux-gnu/libpython3.10.so`.

Reply:
@ncguilbeault I think I got it, this last issue is resolved simply by installing `libpython-dev` with the corresponding python version. With this I was able to get the kernel running with almost no code changes, I will submit an alternative PR for us to review soon.